### ISSUE TITLE
get_document_outline maps expression-bodied constructors to Kind "Constructor" (ExpressionBody / ArrowExpressionClause ignored)

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#713]
## Summary

`get_document_outline` never inspects a walked constructor's **`ExpressionBody`** (`ArrowExpressionClause`) when choosing Kind. A declaration such as `public C(int x) { _x = x; }` and `public C(string s) => _x = s.Length;` both get baseline Kind `"Constructor"` with a bare Name — the same Kind as each other — so clients that key navigation, codegen, or “is expression-bodied” heuristics off outline Kind alone cannot tell an expression-bodied instance constructor from a peer with a block body.

Constructors **are** already walked — so this is a Kind/shape hole on members that *are* outlined, not an omission hole. `#711` covers method **`ExpressionBody` / `ArrowExpressionClause`** and `#710` covers property `ExpressionBody`; `#711`'s Separate follow-ups incorrectly marked constructor expression-body Kind as “N/A in C#”. That claim is wrong: since C# 7.0, **instance** constructors may use expression bodies (`ConstructorDeclarationSyntax.ExpressionBody` is a real Roslyn node). Static constructors cannot be expression-bodied (they require a block body) — out of scope as a distinct Kind axis here (`#645` already covers `StaticKeyword`). No prior ticket covers expression-bodied constructor outline Kind.

## Bug

In [`GetDocumentOutlineOperation`](https://github.com/JoshuaRamirez/RoslynMcpServer/blob/ef31518bf403aa505beedd07b05dd7a9c149b229/src/RoslynMcp.Core/Query/GetDocumentOutlineOperation.cs), walked constructors are outlined without reading `ConstructorDeclarationSyntax.ExpressionBody`:

```csharp
// Constructors (~L147–153)
foreach (var ctor in type.Members.OfType<ConstructorDeclarationSyntax>())
{
    totalCount++;
    children.Add(CreateEntry(ctor.Identifier.Text, "Constructor",
        ctor.GetLocation(), GetAccessibility(ctor.Modifiers)));
}
```

Kind is hard-coded `"Constructor"`; there is **no** inspection of `ctor.ExpressionBody` / `ArrowExpressionClause` when choosing Kind — and Name never embeds the expression either. Confirmed: **zero** references to `ExpressionBody` / `ArrowExpressionClause` anywhere in `GetDocumentOutlineOperation.cs`. Even after `#711` / `#710` land for methods / properties (and sibling constructor modifier tickets land), an expression-bodied instance-ctor peer would still collide on Kind with its block-bodied twin unless this hole is fixed.

For a document that holds:

```csharp
namespace N
{
    public class C
    {
        private readonly int _x;
        public C(int x) { _x = x; }
        public C(string s) => _x = s.Length;
        public C(long y) : this((int)y) => _x = (int)y;
        static C() { }
    }
}
```

`C` (int), `C` (string), `C` (long), and the static ctor all outline as Kind `"Constructor"` (Accessibility / Line / Column may differ; Name is the type identifier for all — `#644` covers ParameterList Name collisions separately). Kind does **not** mark expression-bodied. Clients that key off Kind alone cannot flag expression-bodied constructors; they would have to re-parse source / `ArrowExpressionClause` even though the syntax tree already exposes `ctor.ExpressionBody`. Note: the `C(long)` sample mixes a constructor initializer (`: this(...)`) with an expression body — Kind still has no ExpressionBody signal; ConstructorInitializer Kind refinements are a Separate follow-up if desired.

By contrast, `#711` teaches method ExpressionBody Kind and `#710` property ExpressionBody Kind. Sibling constructor Kind tickets `#645` / `#701` / `#702` / `#707` cover static / unsafe / parameter-modifiers / scoped — none cover **`ExpressionBody` on `ConstructorDeclarationSyntax`**.

## Expected

Outline Kind (and/or Name) for a walked instance `ConstructorDeclarationSyntax` whose `ExpressionBody` is an `ArrowExpressionClause` should be distinguishable from a peer with a block `Body` (and from a peer with neither, e.g. `extern`). Clients must not have to re-parse source or rely solely on Name text to learn “this constructor is expression-bodied” from the outline alone. Ordinary constructors without `ExpressionBody` keep baseline Kind `"Constructor"` (subject to sibling Kind tickets such as `#645` / `#701`). Method / property expression-bodied Kind remain `#711` / `#710`; indexer / operator / destructor / local-function expression-bodied Kind refinements remain Separate follow-ups (walk first where omitted). Static constructors keep block-only C# rules (no ExpressionBody possible) — do not invent a false “expression-bodied static ctor” Kind.

## Actual

On master `ef31518bf403aa505beedd07b05dd7a9c149b229`, the sample above returns outline entries with baseline Kind `"Constructor"` for the block instance ctor, the expression-bodied instance ctors, and the static ctor — Kind alone cannot separate expression-bodied constructors from peers without `ArrowExpressionClause`, and no other outline field carries the expression body.

## Evidence

- File: [`src/RoslynMcp.Core/Query/GetDocumentOutlineOperation.cs`](https://github.com/JoshuaRamirez/RoslynMcpServer/blob/ef31518bf403aa505beedd07b05dd7a9c149b229/src/RoslynMcp.Core/Query/GetDocumentOutlineOperation.cs) (constructors ~L147–153; Kind hard-coded `"Constructor"`; no `ctor.ExpressionBody` / `ArrowExpressionClause` Kind branching)
- Master SHA: `ef31518bf403aa505beedd07b05dd7a9c149b229`
- Repo search: under `GetDocumentOutlineOperation`, zero `ExpressionBody` / `ArrowExpressionClause` references when choosing Kind (or anywhere in the file)
- Language fact: C# 7.0+ allows expression-bodied **instance** constructors (`public C(int x) => _x = x;`). `#711` Separate follow-ups incorrectly said constructor expression-body Kind is N/A — that note is superseded by this issue. Static constructors cannot use expression bodies.
- Tests: `GetDocumentOutlineToolTests.cs` has no constructor / `ExpressionBody` Kind (or Name) assertion
- Dup check: no prior open/closed issue titled for constructor `ExpressionBody` / expression-bodied constructor outline Kind (`#711` / `#710` are method / property only; `#711` wrongly deferred constructors as N/A; `#645`/`#701`/`#702`/`#707` are constructor modifiers / params; `#632` is primary-ctor ParameterList walk omission; `#635` is local functions inside constructors)

## Repro sketch

1. Open a document containing parallel instance constructors with and without expression bodies, e.g. the `namespace N` sample above.
2. Call `get_document_outline` on that document.
3. Observe the block `C(int)`, expression-bodied `C(string)` / `C(long)`, and static `C()` all outlined as baseline Kind `"Constructor"` with bare Name — no Kind signal that the string/long overloads carry `ArrowExpressionClause` (Accessibility / Line / Column do not mark this either; `#644` Name/ParameterList is a separate collision axis).

## Relation

- Twin follow-up correcting `#711` Separate follow-ups (expression-bodied **constructor** Kind after method ExpressionBody `#711` / property ExpressionBody `#710`). Continues **expression-body Kind fidelity** onto **`ConstructorDeclarationSyntax.ExpressionBody` / `ArrowExpressionClause`**. Named naturally after `#711` (same syntax axis, different declaration site — and corrects the mistaken “N/A” deferral). Not blocked on any walk-omission ticket — constructors are already outlined.
- Sibling constructor Kind holes `#645` (static — orthogonal), `#701` (unsafe — orthogonal), `#702` (ref/in/out/params — orthogonal), `#707` (scoped params — orthogonal).
- Sibling ExpressionBody tickets `#710` / `#711` (same axis, different sites — already filed).
- Primary constructor ParameterList `#632` and local-functions-in-ctors `#635` are walk omissions, not Kind on walked `ConstructorDeclarationSyntax` ExpressionBody.
- Destructor ExpressionBody Kind waits on `#631` (destructors not walked yet); indexers / operators wait on `#627` / `#629` / `#630`.
- No open leftover for this track (label **bug** only; open leftover count 0).

## Separate follow-ups

- ConstructorInitializer Kind refinements (`: base(...)` / `: this(...)`) if desired as its own issue (orthogonal syntax axis on the same walked constructors).
- Expression-bodied destructor Kind once `#631` outlines destructors (C# also allows `~C() => ...`).
- Expression-bodied indexer / operator Kind refinements once `#627` / `#629` / `#630` outline those members.
- Expression-bodied local-function Kind once `#633` (and siblings) outline local functions.
- `[Flags]` / attribute-driven Kind refinements if desired as its own issue (prefer syntax-first; attributes are a different axis).
- Shared body-local walker for `#633`–`#639`.
- Primary-constructor ParameterList Kind refinements remain with `#632` (walk omission first).
- Nested-delegate application remains with `#628`.

## Acceptance criteria

- [ ] A walked instance `ConstructorDeclarationSyntax` whose `ExpressionBody` is present (`ArrowExpressionClause`) is distinguishable in the outline from a peer with a block `Body` (and from peers with neither, if applicable) (e.g. Kind forms such as `"ExpressionBodiedConstructor"` / `"ConstructorExpressionBody"`, or equivalently unambiguous documented Name/Kind forms). Document how this composes with sibling constructor Kind tickets (`#645` / `#701` / `#702` / `#707`) under a clear priority (e.g. ExpressionBody vs StaticKeyword vs UnsafeKeyword when more than one could apply — note ExpressionBody does not apply to static constructors in C#).
- [ ] Name / Line / Column / Accessibility must **not** be the only signals — Kind (or documented Name+Kind form) must reflect expression-bodied vs block-bodied when `ArrowExpressionClause` is present (clients must not have to re-parse source / expression bodies to learn this from Kind alone). Ordinary constructors without `ExpressionBody` keep baseline Kind `"Constructor"` (subject to sibling Kind tickets such as `#645`). Method / property ExpressionBody Kind are out of scope (do not redefine in this issue; `#711` / `#710` own those sites).
- [ ] Cover at least: one expression-bodied vs block-bodied instance-constructor pair. Static constructors remain non-expression-bodied (document; do not invent ExpressionBody Kind for them). Optional: one ctor that also has a ConstructorInitializer — ExpressionBody Kind must still apply; ConstructorInitializer Kind is out of scope.
- [ ] Regression test(s) in `GetDocumentOutlineToolTests` (or equivalent) asserting the outline distinguishes at least one expression-bodied instance constructor from a block-bodied peer.
- [ ] Do **not** treat fixing `#711`/`#710` (other ExpressionBody sites), `#645`/`#701`/`#702`/`#707` (constructor modifiers/params), `#632`/`#635`/`#631` (walk omissions), or ConstructorInitializer Kind as in-scope for this issue — though naming should stay consistent with the broader ExpressionBody Kind vocabulary (`#711` / `#710`).
- [ ] Explicitly supersede `#711`'s “constructor expression-body Kind is N/A” Separate follow-up note.

## Suggested fix direction (high-level)

When outlining each `ConstructorDeclarationSyntax`, if `ctor.ExpressionBody` is non-null (an `ArrowExpressionClause`), choose a Kind (and/or Name form) that reflects the expression-bodied shape, composing with any eventual `#645`/`#701`/`#702`/`#707` constructor Kind rules under a documented priority. Keep Line / Column / Accessibility as today. Prefer detecting via syntax (`ctor.ExpressionBody`) rather than requiring a semantic model (the operation is documented as pure syntax walk). Skip inventing ExpressionBody Kind for static constructors (they cannot have `ExpressionBody` in C#). Method / property ExpressionBody Kind stay `#711` / `#710`. Destructor / indexer / operator ExpressionBody Kind wait on their walk tickets.
